TaskHistory.Impl/Tasks/TaskRepo.cs

```diff
@@ -6,77 +6,91 @@
 using TaskHistory.Api.Users;
 using TaskHistory.Impl.MySql;
 using TaskHistory.Impl.Tasks;
+using System.Configuration;
 
 namespace TaskHistory.Impl.Tasks
 {
-	public class TaskRepo : AbstractMySqlRepo, ITaskRepo
+	public class TaskRepo : ITaskRepo
 	{
 		private readonly TaskFactory _taskFactory;
 
 		public ITask InsertNewTask (string taskContent)
 		{
-			var command = _mySqlCommandFactory.CreateMySqlCommand ("Tasks_Insert");
-			command.Parameters.Add (new MySqlParameter ("pTaskContent", taskContent));
-			command.Connection.Open ();
-
-			MySqlDataReader reader = command.ExecuteReader (CommandBehavior.CloseConnection);
-			ITask task = null;
-
-			if (reader.Read ()) 
+			using (var connection = new MySqlConnection (ConfigurationManager.AppSettings ["MySqlConnection"]))
+			using (var command = new MySqlCommand ("Tasks_Insert")) 
 			{
-				task = CreateTaskFromReader (reader);
+				command.CommandType = CommandType.StoredProcedure;
+				
+				command.Parameters.Add (new MySqlParameter ("pTaskContent", taskContent));
+				command.Connection.Open ();
+
+				using (var reader = command.ExecuteReader (CommandBehavior.CloseConnection)) 
+				{
+					ITask task = null;
+
+					if (reader.Read ()) 
+					{
+						task = CreateTaskFromReader (reader);
+					}
+
+					return task;
+				}
 			}
-			command.Connection.Close ();
-
-			return task;
 		}
 			
 		public void DeleteTask (int taskId)
 		{
-			var command = _mySqlCommandFactory.CreateMySqlCommand ("Tasks_LogicalDelete");
-			command.Parameters.Add (new MySqlParameter ("pTaskId", taskId));
-			command.Connection.Open ();
-			command.ExecuteNonQuery ();
-			command.Connection.Close ();
+			using (var connection = new MySqlConnection (ConfigurationManager.AppSettings ["MySqlConnection"]))
+			using (var command = new MySqlCommand ("Tasks_LogicalDelete")) 
+			{
+				command.Parameters.Add (new MySqlParameter ("pTaskId", taskId));
+				command.Connection.Open ();
+				command.ExecuteNonQuery ();
+			}
+
 		}
 
 		public void UpdateTask (ITask newTaskDto)
```

Comment:
Let's not re-use ITask as newTaskDto. 

ITask is an ITask object. The newTaskDto is just a collection of data fields that we need. They are not the same thing.